### PR TITLE
feat: setup initial DA module structure

### PR DESCRIPTION
**Pull request by tkxkd0159**
## Description

Design interfaces and make scratch codes for Data Availability features.
1. Queue transaction must be processed sequentially and within a specific time.
2. Sequencers should include all queue transactions before a synced L1 block height in L2
3. Sequencers should compress their CTC batch data when they submit
4. We assume that CTC is append-only and this data are submitted correctly by a sequencer.
5. SCC guarantees a state transition in L2. SCC Batch matches same index of CTC Batch. CTC data will be finalized after the matching SCC Batch passes the challenge period.

## Checklist:

- [ ] I followed the [contributing guidelines](https://github.com/Finschia/finschia-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/Finschia/finschia-sdk/blob/main/CODE_OF_CONDUCT.md).
- [ ] I have added a relevant changelog to `CHANGELOG.md`
- [ ] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

File: x/or/da/keeper/cc.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package keeper
+
+func (k Keeper) appendSequencerBatch() {
```
Comment (torao) [inline x/or/da/keeper/cc.go:3]: It's better to leave some comment on the lint check failure, as similar check failures may be reported in another PR later on.

**Review by torao — APPROVED** [review]

File: proto/finschia/or/da/v1/da.proto [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+  // Assumed to contain all contiguous batches between base and height (inclusive).
+  uint64 base = 1;
+  uint64 height = 2;
+
+  // The name of rollup chain (use as key).
+  bytes name = 3;
+
+  // The latest batch timestamp when the sequencer submits.
+  google.protobuf.Timestamp last_sequencer_submit = 4 [(gogoproto.stdtime) = true, (gogoproto.nullable) = false];
+}
```
Comment (ulbqb) [inline proto/finschia/or/da/v1/da.proto:166]:
```suggestion
}

```

File: proto/finschia/or/da/v1/genesis.proto [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+message CC {
+  CCState state = 1;
+  repeated CCRef history = 2;
+}
+message SCC {
+  SCCState state = 1;
+  repeated SCCRef history = 2;
+}
```
Comment (ulbqb) [inline proto/finschia/or/da/v1/genesis.proto:24]:
```suggestion
}

```

File: proto/finschia/or/da/v1/query.proto [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import "finschia/or/da/v1/params.proto";
+import "finschia/or/da/v1/da.proto";
+
+option go_package = "github.com/Finschia/finschia-sdk/x/or/da/types";
+
+// Query defines the gRPC querier service.
+service Query {
+  // Parameters queries the parameters of the module.
+  rpc Params(QueryParamsRequest) returns (QueryParamsResponse) {
+    option (google.api.http).get = "/checkers/da/params";
```
Comment (ulbqb) [inline proto/finschia/or/da/v1/query.proto:16]:
```suggestion
    option (google.api.http).get = "/finschia/or/da/v1/params";
```

File: proto/finschia/or/da/v1/query.proto [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+message QueryCCBatchRequest {
+  string rollup_name = 1;
+  uint64 batch_height = 2;
+}
+
+message QueryCCBatchResponse {
+  CCRef batch_loc = 1;
+}
```
Comment (ulbqb) [inline proto/finschia/or/da/v1/query.proto:54]:
```suggestion
}

```

File: proto/finschia/or/da/v1/tx.proto [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+message MsgAppendSCCBatchResponse {}
+
+message MsgRemoveSCCBatch {
+  string rollup_name = 1;
+  uint64 batch_index = 2;
+}
+
+message MsgRemoveSCCBatchResponse {}
```
Comment (ulbqb) [inline proto/finschia/or/da/v1/tx.proto:68]:
```suggestion
message MsgRemoveSCCBatchResponse {}

```

File: proto/finschia/or/da/v1/query.proto [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+// Query defines the gRPC querier service.
+service Query {
+  // Parameters queries the parameters of the module.
+  rpc Params(QueryParamsRequest) returns (QueryParamsResponse) {
+    option (google.api.http).get = "/checkers/da/params";
+  }
+
+  rpc CCBatches(QueryCCBatchesRequest) returns (QueryCCBatchesResponse) {
+    option (google.api.http).get = "/checkers/da/batches/cc";
```
Comment (ulbqb) [inline proto/finschia/or/da/v1/query.proto:20]:
```suggestion
    option (google.api.http).get = "/finschia/or/da/v1/batches/cc";
```

**Review by ulbqb — CHANGES_REQUESTED** [review]

**Review by ulbqb — APPROVED** [review]

**Pull request merged**